cpp/include/DataStorm/Types.h

```diff
@@ -240,58 +240,6 @@ namespace DataStorm
         static T decode(const Ice::CommunicatorPtr& communicator, const Ice::ByteSeq& value) noexcept;
     };
 
-    /**
-     * The Cloner template provides a method to clone user types.
```

Comment:
Removed Cloner as it was used/tested only for "class by value". Its documentation (in the IceStorm manual) is also incomplete and rather confusing/incorrect.

Reply:
According to the documentation, the cloner template is also used by partial updates for cloning types. I believe it is required for class-by-reference scenarios to ensure `ice_clone` is called instead of using the shared_ptr copy constructor.

It might be worth updating `DataStorm/partial` to test class-by-reference behavior, as the current tests only cover structs. The original implementation tested a class-by-value, which is a removed by this PR.

From my understanding, without this change, the partial update will modify the previous value directly rather than computing a new value from previous.

Reply:
You were right, it was used by the partial feature.

I restored Cloner and updated the partial test to use class-by-reference.